sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -524,14 +635,150 @@ protected EventProcessor()
+                // and surrender ownership.
+
+                var stopPartitionProcessingTasks = ActivePartitionProcessors.Keys
+                    .Select(partitionId => StopProcessingPartitionAsync(partitionId, ProcessingStoppedReason.Shutdown, CancellationToken.None))
+                    .ToArray();
+
+                if (async)
+                {
+                    await Task.WhenAll(stopPartitionProcessingTasks).ConfigureAwait(false);
+                    await LoadBalancer.RelinquishOwnershipAsync(cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    Task.WaitAll(stopPartitionProcessingTasks);
+
+#pragma warning disable AZC0102 // Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.
+                    LoadBalancer.RelinquishOwnershipAsync(cancellationToken).GetAwaiter().GetResult();
+#pragma warning restore AZC0102 // Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.
+                }
+
+                // Dispose of the processing task and reset processing state to
+                // allow the processor to be restarted when this method completes.
+
+                _runningProcessorTask.Dispose();
+                _runningProcessorTask = null;
+            }
+            catch (Exception ex)
+            {
+                Logger.EventProcessorStopError(Identifier, EventHubName, ConsumerGroup, ex.Message);
+                throw;
             }
+            finally
+            {
+                Logger.EventProcessorStopComplete(Identifier, EventHubName, ConsumerGroup);
+
+                // If the cancellation token was signaled during the attempt to acquire the
+                // semaphore, it cannot be safely released; ensure that it is held.
+
+                if (releaseSync)
+                {
+                    ProcessorRunningSync.Release();
+                }
+            }
+
+            // Surface any exception that was captured when the processing task was
+            // initially awaited.
 
+            if (processingException != default)
+            {
+                throw processingException;
+            }
+        }
+
+        /// <summary>
+        ///   Performs the tasks needed to execute processing for this <see cref="EventProcessor{TPartition}" /> instance, managing owned partitions and
+        ///   load balancing between associated processors.
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        private async Task RunProcessingAsync(CancellationToken cancellationToken)
+        {
+            CreateConnection();
+            await Task.Delay(Timeout.Infinite, cancellationToken).ConfigureAwait(false);
```

Comment:
Just curious - is this still a work in progress?  If not - what's the point of this delay?

Reply:
Yup; this is the next set of functionality to be implemented.    Currently, it's just mimicking the long running process by blocking until cancellation is triggered, as the actual processing task will do.